go-mode.el

```diff
@@ -1470,24 +1470,26 @@ description at POINT."
       (error "Cannot use godef on a buffer without a file name")
     (let ((outbuf (generate-new-buffer "*godef*"))
           (coding-system-for-read 'utf-8)
-          (coding-system-for-write 'utf-8))
-      (prog2
-          (call-process-region (point-min)
-                               (point-max)
-                               godef-command
-                               nil
-                               outbuf
-                               nil
-                               "-i"
-                               "-t"
-                               "-f"
-                               (file-truename (buffer-file-name (go--coverage-origin-buffer)))
-                               "-o"
-                               ;; Emacs point and byte positions are 1-indexed.
-                               (number-to-string (1- (position-bytes point))))
-          (with-current-buffer outbuf
-            (split-string (buffer-substring-no-properties (point-min) (point-max)) "\n"))
-        (kill-buffer outbuf)))))
+          (coding-system-for-write 'utf-8)
+          (filename (file-truename (buffer-file-name (go--coverage-origin-buffer))))
+          (token-position (number-to-string (1- (position-bytes point)))))
+      (if (tramp-tramp-file-p filename)
+          (with-parsed-tramp-file-name filename nil
```

Comment:
You don't need to use TRAMP functions here.

- Use `file-local-name` to get the local name of a remote file.
- Use `(concat (file-name-remote-p default-directory) filename)` to convert local names on a remote system to remote names.

That way the patch works for all special file names, not just TRAMP.